fairscale/optim/oss.py

```diff
@@ -65,6 +76,7 @@ def __init__(self, params: _params_t, optim: Type[Optimizer] = SGD, group: Any =
 
         # Current device is set by the parameters allocated to this rank
         self._device = self.partition_parameters()[self.rank][0]["params"][0].device
+        self._buffer = torch.zeros(buffer_size).to(self._device)
```

Comment:
@msbaines  could you elaborate on the batch buffer which needs to be per-device ? I'm guessing given your comment that this is not enough, but it's not obvious to me

Reply:
The parameters could be stored on multiple devices (e.g. gpipe) so we need a batch buffer for each device.

Reply:
> The parameters could be stored on multiple devices (e.g. gpipe) so we need a batch buffer for each device.

See oss_ddp.py